front/src/Connexion/ConnectionManager.ts

```diff
@@ -107,10 +112,14 @@ class ConnectionManager {
                 this.localUser = localUser;
                 localUserStore.saveUser(localUser);
             }
-            return Promise.resolve(room);
+        }
+        if (room == undefined) {
+            return Promise.reject(new Error("Invalid URL"));
         }
 
-        return Promise.reject(new Error("Invalid URL"));
+        console.log("initGameConnexion => serviceWorker", room);
+        this.serviceWorker = new _ServiceWorker();
```

Comment:
are you sure you need to create the service worker here? Service workers are not supposed to interact directly with the rest of the codebase. Maybe create it in a separate file?